projects/igniteui-angular/src/lib/grids/cell.component.ts

```diff
@@ -210,10 +210,11 @@ export class IgxGridCellComponent implements OnInit, OnChanges, OnDestroy, CellT
      *
      * @memberof IgxGridCellComponent
      */
-    public get context(): any {
+    public get context(): IgxCellTemplateContext {
         const ctx = {
```

Comment:
Alternate implementation to avoid setting a fake value in the ctx initialization.

```typescript
        const getCellType = () => this.getCellType(true);
        const ctx = {
            $implicit: this.value,
            additionalTemplateContext: this.column.additionalTemplateContext,
            get cell() {
                /* Turns the `cell` property from the template context object into lazy-evaluated one.
                 * Otherwise on each detection cycle the cell template is recreating N cell instances where
                 * N = number of visible cells in the grid, leading to massive performance degradation in large grids.
                 */
                return getCellType();
            }
        };
        if (this.editMode) {
            ctx['formControl'] = this.formControl;
        }
        if (this.isInvalid) {
            ctx['defaultErrorTemplate'] = this.defaultErrorTemplate;
        }
        return ctx;
```

Reply:
Yes, good point. Thanks, Pablo!